### PR TITLE
docs/provider: Update contributing guide with Resource Contribution Guidelines section

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Replaces existing Terraform Schema and Code Idiosyncracies section with many additional common code review items.

### CODE REVIEW

**Comment by ewbankkit** [comment]
Great to have these codified, I always forget a different few of these every time 😄.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -30,9 +30,9 @@ we need to be able to review and respond quickly.
         - [Documentation Update](#documentation-update)
         - [Enhancement/Bugfix to a Resource](#enhancementbugfix-to-a-resource)
         - [New Resource](#new-resource)
+        - [New and Updated Resource Guidelines](#new-and-updated-resource-guidelines)
```
[the remark this answers is not in the document]
Comment (ryndaniels) [reply .github/CONTRIBUTING.md]: I like "Resource Contributing Guidelines" as well - for some reason my brain keeps parsing "New and Updated Resource Guidelines" as "the guidelines are new/updated"

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -260,6 +265,72 @@ existing resources, but you still get to implement something completely new.
    folder. This is to avoid conflicts as the vendor versions tend to be fast
    moving targets.
 
+#### New and Updated Resource Guidelines
+
+The Terraform AWS Provider follows common practices to ensure consistent and reliable implementations across all resources in the project. While there may be older schema and resource code present that predates these guidelines, new submissions are generally expected to adhere to these items to maintain Terraform Provider quality. For any guidelines listed, contributors are encouraged to ask any questions and community reviewers are encouraged to provide review suggestions based on these guidelines to speed up the review and merge process.
+
+The below are required items that will be noted during submission review and prevent immediate merging:
```
Comment (ryndaniels) [inline .github/CONTRIBUTING.md]: Does "prevent immediate merging" here mean that immediate merging will be prevented *if* any of these required items are missing? That's been my assumption but the phrasing is a bit ambiguous.

File: .github/CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (bflad) [reply .github/CONTRIBUTING.md]: Thanks! I'm going to take both of these suggestions and incorporate them into an update shortly. 👍

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (bflad) [reply .github/CONTRIBUTING.md]: Words are hard! What we are trying to relay to contributors here is that these items, if applicable, would be required feedback. We occasionally receive contributions where folks are curious why it is not merged yet and this list would be a reference for those situations. If you have better suggestions for the wording, we can definitely change this. 😄

File: .github/CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (bflad) [reply .github/CONTRIBUTING.md]: Updated -- please take a look

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (tombuildsstuff) [reply .github/CONTRIBUTING.md]: Perhaps:
> the following items, if applicable, are required for new/updates to existing resources?

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -323,12 +323,10 @@ manually sourced values from documentation.
 
 ### Common Review Items
 
-The sections below outline common pull request feedback items that may be helpful to contributors to know before submission and during the review process.
+The Terraform AWS Provider follows common practices to ensure consistent and reliable implementations across all resources in the project. While there may be older resource and testing code that predates these guidelines, new submissions are generally expected to adhere to these items to maintain Terraform Provider quality. For any guidelines listed, contributors are encouraged to ask any questions and community reviewers are encouraged to provide review suggestions based on these guidelines to speed up the review and merge process.
```
Comment (nywilken) [inline .github/CONTRIBUTING.md]: :100:

File: .github/CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (nywilken) [reply .github/CONTRIBUTING.md]: Thanks for making the update. This looks great to me :+1:

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (nywilken) [inline .github/CONTRIBUTING.md]:
```suggestion
The following resource checks need to be addressed before your contribution can be merged. The exclusion of any applicable check may result in a delayed time to merge. 
```

**Pull request merged**